packages/viewer/src/mathml.ts

```diff
@@ -66,22 +66,26 @@ export async function renderMathML(properties: Properties, root: HTMLElement): P
 
   for(const mathElement of [...root.getElementsByTagName('math')]) {
     const mml = htmlEntitiesToXmlEntities(mathElement.outerHTML);
+    try {
+      let result;
+      
+      // Transform mml to img.
+      if (properties.wirispluginperformance === 'true') {
+        result = await showImage(mml, properties.lang, properties.editorServicesRoot, properties.editorServicesExtension);
+      } else {
+        result = await createImage(mml, properties.lang, properties.editorServicesRoot, properties.editorServicesExtension);
+      }
 
-    let result;
+      // Set img properties.
+      const img = await setImageProperties(properties, result, mml);
+      // const fragment = document.createRange().createContextualFragment(data.result.content);
 
-    // Transform mml to img.
-    if (properties.wirispluginperformance === 'true') {
-      result = await showImage(mml, properties.lang, properties.editorServicesRoot, properties.editorServicesExtension);
-    } else {
-      result = await createImage(mml, properties.lang, properties.editorServicesRoot, properties.editorServicesExtension);
+      // Replace the MathML for the generated formula image.
+      mathElement.parentNode?.replaceChild(img, mathElement);
+    } catch {
+      console.log(`Cannot render ${mml}: invalid MathML format.`);
```

Comment:
Maybe `console.error` would be more appropiated in this case!

Reply:
Done!